### PR TITLE
Add basic progressbar to public upload page

### LINKED ISSUE #4164: Add an upload progress bar in "File drop (upload only)" shares
### Feature request

When uploading a big file in a "File drop (upload only)" share, we have no idea of the remaining upload time.
It would be a great enhancement to add a progress bar +remaining upload time, like in other uploads.

### PR DESCRIPTION
**Pull request by CHRINIMUE**
I started to add a progress bar to the public upload page (files_sharing)

I have to do some more design adjustments but then it should be ready.

Fix: #4164

(This is my first pull request for the nextcloud server, so i don't know how the workflow is here exactly... how can I add a label?)

### CODE REVIEW

**Comment by gary-kim** [comment]
> (This is my first pull request for the nextcloud server, so i don't know how the workflow is here exactly... how can I add a label?)

Someone in the org has to do it. You should be sent an invite once your first pull request gets merged :). Ping me when it's ready for review and I'll send review requests for it.

File: apps/files_sharing/js/files_drop.js [hunk 1]
```diff
@@ -89,12 +90,14 @@
 				sequentialUploads: true,
 				start: function(e) {
 					self._uploading = true;
+					$('#drop-upload-status').text('starting');
```
Comment (juliushaertl) [inline apps/files_sharing/js/files_drop.js]: Please make those texts translatable https://docs.nextcloud.com/server/latest/developer_manual/app/view/l10n.html?highlight=translations#javascript

File: apps/files_sharing/js/files_drop.js [hunk 2]
```diff
@@ -124,7 +127,17 @@
 						$('#drop-upload-done-indicator').addClass('hidden');
 						$('#drop-upload-progress-indicator').removeClass('hidden');
 					}
-				}
+				},
+				progress: function (e, data) {
+					var progress = parseInt(data.loaded / data.total * 100, 10);
+					if(progress === 100) {
+                                                $('#drop-upload-progress-bar').val(100);
```
Comment (juliushaertl) [inline apps/files_sharing/js/files_drop.js]: We use tabs for indentation :wink:

File: apps/files_sharing/js/files_drop.js (hunk 1 above)
Comment (juliushaertl) [reply apps/files_sharing/js/files_drop.js]: Nevermind, just seen you are still on it :+1:

Comment (CHRINIMUE) [reply apps/files_sharing/js/files_drop.js]: Yes... just did it 😄

Comment (CHRINIMUE) [reply apps/files_sharing/js/files_drop.js]: But I hope I did it in the right way? The new strings will automatically added to Transifex? Or how's that working? 🤔

File: apps/files_sharing/js/files_drop.js (hunk 2 above)
Comment (CHRINIMUE) [reply apps/files_sharing/js/files_drop.js]: Oh, wow. Yes... that should not happen 😆

**Comment by CHRINIMUE** [comment]
@gary-kim Ok, now I finished it...  😅

**Comment by gary-kim** [comment]
Short gif for reference:
output

**Review by jancborchardt — APPROVED** [review]
Looks great! Thanks a lot @CHRINIMUE for the contribution! (And thank you @gary-kim for the gif for easy review!)

**Review by gary-kim — COMMENTED** [review]
Just something that maybe could be good to give some thought to. When you upload a lot of files at once, the list just keeps on growing so it gets a bit long. This becomes a bit of a problem since new files appear at the bottom of the list.
output

**Review by juliusknorr — APPROVED** [review]
Tested and works like a charm. Really nice improvement. :+1:

**Comment by jancborchardt** [comment]
> Just something that maybe could be good to give some thought to. When you upload a lot of files at once, the list just keeps on growing so it gets a bit long. This becomes a bit of a problem since new files appear at the bottom of the list.

Good point. Let’s add new uploads to the top of the list instead? :)

**Comment by CHRINIMUE** [comment]
Mhhh... yes. A good point... 

There is another old problem: uploading two (or more) same named files. If one is uploaded, all other are are shown as finished... but at this moment they are not finished.

But I would say, I can fix that in another PR?

**Comment by jancborchardt** [comment]
> But I would say, I can fix that in another PR?

Yep, absolutely! :) All that is stuff for follow-ups.

**Comment by zachmarion** [comment]
This is awesome! Thank you so much for this feature. Please forgive my newb question, but if I want to integrate this into my system right now, would I have to manually copy all of the committed files to their correct directories?

**Comment by jancborchardt** [comment]
@zachmarion this fix should be in the current Nextcloud release, version 19. :) Updating to that will do the trick. Note that maybe you are not getting the indication of a possible update yet since we do a staged rollout – maybe it will only come as soon as we released 19.0.1 or similar.

**Comment by zachmarion** [comment]
I had upgraded to version 18. I just changed my Update channel to Beta and went to 19. It works and looks great! This is very helpful. I am using Nextcloud for video production, and I'm facilitating a lot of remote self-tapes these days because of the pandemic. When I upload the file to myself from the remote machine, it is super helpful to see the progress. Thank you again for all of your hard work on such a crucial tool for my business. If there is a way to support your work in a more meaningful way, let me know!

**Comment by jancborchardt** [comment]
Thanks for the kind words @zachmarion! Happy to hear that Nextcloud is useful to you. :)

> If there is a way to support your work in a more meaningful way, let me know!

Yes, if you would like to support the project there are some main ways:
- Getting an Enterprise Subscription (probably too much if you mainly use it for yourself): https://nextcloud.com/pricing/
- Donate to our Nextcloud Include program: https://nextcloud.com/include/ – just let us know if you would like to donate and we can give you a Paypal address (donations are handled via the non-profit KDE e.V., cc @ChristophWurst for reference)
- Of course it’s always appreciated to help other people out on the forums: https://help.nextcloud.com

**Pull request merged**